torch/testing/_internal/composite_compliance.py

```diff
@@ -0,0 +1,172 @@
+import torch
+import contextlib
+from typing import Iterator
+from torch.utils._pytree import tree_map
+from functools import partial
+from torch.utils._python_dispatch import enable_python_mode
+
+# TODO: move this into library proper
```

Comment:
NOPE you get to fix this TODO 🤪

Reply:
torch/overrides.py is reasonable; otherwise we should make a new toplevel for dispatch-y related functionality

Reply:
Lol this is the 2nd time I've copied this function

Reply:
Do we really want to make this a thing? It doesn't feel like the right fix for this problem.
If people start to use this, then all the nice use case from https://colab.research.google.com/drive/12FN24CPRe_FVxjE46SZTu8q3j4RVSq15#scrollTo=XVeRMjCT8ptG would break because the other subclass would never be called...

Reply:
So @zou3519 talked a while back about a more surgical variant of this that simply removes a class from consideration. I'm not sure this is still the right thing to do. In general, some tensors will want a way to say "treat me like my superclass without unwrapping"; where a super() call is inappropriate.

Reply:
Actually I was just debugging something with Anjali related to torch_function and using disable torch function that was creating subtle issues when subclassing.
In this case, proper use of super() is doing the right thing because we go all the way to the `Tensor.__torch_function__` when everything else has been handled and we're good to go.
We really need a mechanism to be able to do the same in torch_dispatch no?